### PR TITLE
Tables cell and checkbox accessibility

### LINKED ISSUE #792: Associate data cells in tables with both the column and row header
### Issue Description

The current django admin tables are set so their labels only include row header, but the 508 compliance requirement is for column headers (with or without row headers) to be present on tables. 

AC
-all data table cells, on all django admin pages are programmatically associated with relevant headers (both column and row)
-using a screen reader this label should be readable

### Additional Context (optional)

see the 508 compliance outcome in #788 

### Issue Links

Relates to: #788 
Relates to: #752
Relates to: #793

### LINKED ISSUE #793: Add accessibility labels to django-admin checkboxes
### Issue Description

Issue Description
The current django admin checkboxes on list views do not have an accessible name and associated label and fail a 508 compliance check. 

AC
- checkboxes on django admin need to have an accessible name and an associated label.
- using a screen reader this accessible name should be readable.

### Additional Context (optional)

see the 508 compliance outcome in #788

### Issue Links

Issue Links
Relates to: #792
Relates to: #788
Relates to: #752

### PR DESCRIPTION
**Pull request by rachidatecs**
## 🗣 Description ##

- Resolve table cell accessibility in django admin by adding a thead with a header row
- Resolve checkbox accessibility in django admin by adding hidden labels to all checkboxes
- Unit tests
- Light CSS to ensure table layout looks OK
- Document ANDI usage to test pages in the browser

Screen Shot 2023-07-26 at 4 06 29 PM

## 💭 Motivation and context ##

Resolves #792, Resolves #793 

## ANDI tests ##

Screen Shot 2023-07-26 at 4 11 28 PM

Screen Shot 2023-07-26 at 4 12 23 PM

### CODE REVIEW

File: docs/developer/README.md [hunk 1]
```diff
@@ -166,6 +166,16 @@ from .common import less_console_noise
             # <test code goes here>
 ```
 
+### Accessibility Testing in the browser
+
+We use the [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) browser extension 
+from ssa.gov for accessibility testing outside the pipeline. 
+
+ANDI will get blocked by our CSP settings, so you will need to install the 
+[Disable Content-Security-Policy extension](https://chrome.google.com/webstore/detail/disable-content-security/ieelmcmcagommplceebfedjlakkhpden) and activate it for the page you'd like
```
Comment (abroddrick) [inline docs/developer/README.md:175]: okay i downloaded the extension and it looks like its activating and then I click andi but still don't see ANDI popping up correctly

File: docs/developer/README.md [hunk 2]
```diff
@@ -203,7 +213,7 @@ Assets are stored in `registrar/assets` during development and served from `regi
 
 We utilize the [uswds-compile tool](https://designsystem.digital.gov/documentation/getting-started/developers/phase-two-compile/) from USWDS to compile and package USWDS assets.
 
-## Making and view style changes
+## Making and viewing style changes
```
Comment (abroddrick) [inline docs/developer/README.md:216]: finally a spelling/grammar issue that isn't my fault

File: src/registrar/assets/sass/_theme/_cisa_colors.scss
Comment (abroddrick) [inline src/registrar/assets/sass/_theme/_cisa_colors.scss:1]: why the name change?

File: src/registrar/assets/sass/_theme/styles.scss [hunk 3]
```diff
@@ -9,3 +9,7 @@
 /*--------------------------------------------------
 --- Custom Styles ---------------------------------*/
 @forward "uswds-theme-custom-styles";
+
+/*--------------------------------------------------
+--- Admin ---------------------------------*/
+@forward "admin";
```
Comment (abroddrick) [inline src/registrar/assets/sass/_theme/styles.scss:15]: endline is off here. running black . should format it, but I also thought Brandon added something to auto add the newlines when needed.

**Review by abroddrick — COMMENTED** [review]
See my comment about still not getting ANDI to work, I want to make sure that works and everything shows up as expected before approving. Also check the endlines at the end of a few of the documents, it's showing the github warning.

**Comment by abroddrick** [comment]
I'm using just my normal mac laptop and this is the view I see, notice the Add button and  view/change section at the top. 
image

Also fyi, when a PR resolves two tickets, it will only link and auto close the PR if you do "resolves #" followed by the number it resolves. I edited your description so that both tickets are auto linked.

File: docs/developer/README.md (hunk 1 above)
Comment (rachidatecs) [reply docs/developer/README.md:175]: Try refreshing with dev tools open > network > no cache, click on ANDI again.

File: src/registrar/assets/sass/_theme/_cisa_colors.scss
Comment (rachidatecs) [reply src/registrar/assets/sass/_theme/_cisa_colors.scss:1]: scss partials that do not have an underscore will compile into their own CSS files. That's not needed for this partial that's architectural and feeds into the scss endpoint.

**Comment by rachidatecs** [comment]
Possibly a frontend caching issue. Try clearing cache and see if that resolves that vertical A D D

**Comment by abroddrick** [comment]
> Possibly a frontend caching issue. Try clearing cache and see if that resolves that vertical A D D

Yep you're right, weird but this solved this issue.

File: docs/developer/README.md (hunk 1 above)
Comment (abroddrick) [reply docs/developer/README.md:175]: I thought I replied to this last night, I tried this. I don't see "no cache" but i clicked "disable cache", refreshed and ran again. I still don't see ANDI opening

File: src/registrar/assets/sass/_theme/styles.scss (hunk 3 above)
Comment (rachidatecs) [reply src/registrar/assets/sass/_theme/styles.scss:15]: Neither did anything. Maybe scss is ignored? Anyways, I'm adding an endline manually.

File: docs/developer/README.md (hunk 1 above)
Comment (abroddrick) [reply docs/developer/README.md:175]: Yeah I already refreshed after disabling the cahce and went ahead and cleared the cache. Still not working

**Comment by Katherine-Osos** [comment]
@rachidatecs The accessibility updates look good to me.

**Review by abroddrick — APPROVED** [review]
Looks perfect! Thanks Rachid!

**Pull request merged**